core/src/main/kotlin/xtdb/compactor/SegmentMerge.kt

```diff
@@ -76,48 +78,120 @@ internal class SegmentMerge(private val al: BufferAllocator) : AutoCloseable {
 
     private val outWriters = OutWriters(al)
 
-    private fun MergePlanTask.merge(outWriter: OutWriter, pathFilter: ByteArray?) {
+    private fun resolveSameSystemTimeEvents(dataReader: RelationReader, path: ByteArray) : RelationReader {
```

Comment:
This does a full copy of the data which adds of course memory pressure.

Reply:
I can't make a call on this one without knowing _how much_ memory pressure we're talking - is it keeping 2 pages in memory rather than one? or do we now need to bring the whole file into memory again?